lib/ide/src/controller/module.rs

```diff
@@ -119,27 +137,57 @@ shared! { Handle
             }
             Ok(())
         }
+
+        /// Get subscriber receiving notifications about changes in module's text representation.
+        pub fn subscribe_text_notifications(&mut self) -> Subscriber<notification::TextChanged> {
+            self.text_notifications.subscribe()
+        }
+
+        /// Get subscriber receiving notifications about changes in module's graph representation.
+        pub fn subscribe_graph_notifications(&mut self) -> Subscriber<notification::GraphChanged> {
+            self.graph_notifications.subscribe()
+        }
     }
 }
 
 impl Handle {
     /// Create a module controller for given location.
     ///
     /// It may wait for module content, because the module must initialize its state.
-    pub async fn new(location:Location, mut file_manager:fmc::Handle, mut parser:Parser)
+    pub async fn new(location:Location, mut file_manager:fmc::Handle, parser:Parser)
     -> FallibleResult<Self> {
-        let logger  = Logger::new(format!("Module Controller {}", location));
+        let logger              = Logger::new(format!("Module Controller {}", location));
+        let path                = location.to_path();
+        let ast                 = Ast::new(ast::Blank{},None);
```

Comment:
And indeed we do: we get the file contents before creating the controller (at least using the "official" constructors). But this new creates "uninitialized", because we will also need "(re)load_module" functionality, and this is for not duplicating code.

Reply:
Creating an empty Module manually is perfectly fine.